app/code/community/Segment/Analytics/Helper/Data.php

```diff
@@ -8,19 +8,45 @@ public function getWriteKey() {
 
     public function getCustomerInformation($data) {
         $fields = trim(Mage::getStoreConfig('segment_analytics/options/customer_traits'));
-        $to_send = preg_split('%[\n\r]%', $fields, -1, PREG_SPLIT_NO_EMPTY);        
-        
+        $to_send = preg_split('%[\n\r]%', $fields, -1, PREG_SPLIT_NO_EMPTY);
+
+        $optionAttrs = Mage::getModel('eav/entity_attribute') ->getCollection();
+        $optionAttrs->addFieldToFilter('entity_type_id', 1)
+            ->getSelect()
+            ->joinLeft(
+                ['ao' => $optionAttrs->getTable('eav/attribute_option')],
+                'ao.attribute_id = main_table.attribute_id',
+                'option_id')
+            ->group('main_table.attribute_id')
+            ->where('(main_table.frontend_input = ? AND ao.option_id > 0)', 'select');
+        $attrByCode = [];
+        foreach ($optionAttrs as $attr) {
+            $attrByCode[$attr->getAttributeCode()] = $attr;
+        }
+
         $data_final = array();
         foreach($to_send as $field) {
-            $data_final[$field] = array_key_exists($field, $data) ? $data[$field] : null;
+            $fieldData = array_key_exists($field, $data) ? $data[$field] : null;
+            if (isset($attrByCode[$field])) {
+                $attr = $attrByCode[$field];
+                $collection = Mage::getResourceModel('eav/entity_attribute_option_collection')
```

Comment:
I don't like doing a second db query inside a loop but not sure how to package this all up nicely in the query above. any suggestions?